src/slurm_plugin/slurm_resources.py

```diff
@@ -149,48 +150,57 @@ def is_drained(self):
         drained(sinfo) is equivalent to IDLE+DRAIN(scontrol) or DOWN+DRAIN(scontrol)
         """
         return self._is_drain() and (
-            self.SLURM_SCONTROL_IDLE_STATE in self.state or self.SLURM_SCONTROL_DOWN_STATE in self.state
+            any(idel_state in self.states for idel_state in self.SLURM_SCONTROL_IDLE_STATE) or self.is_down()
         )
 
     def is_powering_down(self):
         """Check if slurm node is in powering down state."""
-        return self.SLURM_SCONTROL_POWERING_DOWN_STATE in self.state
+        return any(
+            powering_down_state in self.states for powering_down_state in self.SLURM_SCONTROL_POWERING_DOWN_STATES
+        )
 
     def is_power(self):
         """Check if slurm node is in power state."""
-        return self.SLURM_SCONTROL_POWER_STATE == self.state
+        return all(power_state in self.states for power_state in self.SLURM_SCONTROL_POWER_STATE.split("+"))
```

Comment:
isn't there a problem in case self.states contains additional states such as NOT_RESPONDING?
either we leave the string match here or we check for set equality something like set(self.SLURM_SCONTROL_POWER_STATE.split("+")) == self.states (if it works)

Reply:
Change back to check `self.SLURM_SCONTROL_POWER_STATE == self.state`